scripts/convert-to-optional.js

```diff
@@ -0,0 +1,252 @@
+const babel = require("@babel/core");
+const fs = require("fs");
+const path = require("path");
+
+/**
+ * Reference on AST types:
+ * - Docs on babel types https://babeljs.io/docs/babel-types
+ * - Discover and play with AST https://astexplorer.net/ (select @babel/parser as parser)
+ */
+
+/**
+ * Identifies the `[] | [T]` pattern
+ */
+const isOptionalUnionType = (annotation) => {
+  if (annotation.typeAnnotation.type === "TSUnionType") {
+    const types = annotation.typeAnnotation.types;
+    const hasEmptyTuple = types.some(
+      (type) => type.type === "TSTupleType" && type.elementTypes.length === 0
+    );
+    const hasOneElementTyple = types.some(
+      (type) => type.type === "TSTupleType" && type.elementTypes.length === 1
+    );
+    return types.length === 2 && hasEmptyTuple && hasOneElementTyple;
+  }
+};
+
+const EXTERNAL_TYPES = ["Principal", "Uint8Array"];
+
+const mappedInterfaces = new Map();
+
+const transformOptionalTuple = () => {
+  return {
+    visitor: {
+      TSInterfaceDeclaration: (path) => {
+        const name = path.node.id.name;
+        if (name.startsWith("Optional") || name.startsWith("_")) {
+          return;
+        }
+        const t = babel.types;
+        const convertTypeToOptional = (node) => {
```

Comment:
Because this function looks pure, can it be moved outside of the `transformOptionalTuple` function?